torch/csrc/autograd/python_function.cpp

```diff
@@ -743,6 +743,25 @@ PyObject *THPFunction_saved_variables(THPFunction *self, void *_unused)
   END_HANDLE_TH_ERRORS
 }
 
+PyObject *THPFunction_raw_saved_tensors(THPFunction *self, void *_unused)
+{
+  HANDLE_TH_ERRORS
+  auto& saved_variables = self->saved_variables;
+  if (saved_variables.empty())
+    return PyTuple_New(0);
+  int num_saved = saved_variables.size();
+  THPObjectPtr saved(PyTuple_New(num_saved));
+  if (!saved)
+    return nullptr;
+  for(const auto i : c10::irange(num_saved)) {
+    py::object obj = py::cast(saved_variables[i], py::return_value_policy::reference);
```

Comment:
This should probably be abstracted away into another function

Reply:
@soulitzer This snippet code is used three times but not sure where the best place to put it would be

Reply:
Where else is it being used?

Reply:
From here https://pybind11.readthedocs.io/en/stable/advanced/functions.html using py::return_value_policy::reference might lead to undefined behavior if the c++ side is freed and we keep using it on the Python side

Reply:
They are used [here](https://github.com/pytorch/pytorch/pull/60565/files#diff-2bf4105c8354f4a2656b421e0dc44d72542efdde41070ee67a81c7d99c34939aR216) and [here](https://github.com/pytorch/pytorch/pull/60565/files#diff-2bf4105c8354f4a2656b421e0dc44d72542efdde41070ee67a81c7d99c34939aR201)